docs/rules/valid-describe.md

```diff
@@ -0,0 +1,53 @@
+# Enforce valid `describe()` callback (valid-describe)
```

Comment:
I settled on calling this `valid-describe` since this addresses a few different cases. This follows the convention of the `valid-expect` rule. Open to other naming suggestions too!